mlir/test/Ion/PulseOp.mlir

```diff
@@ -0,0 +1,71 @@
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
+        detuning=11.11,
+        polarization=dense<[0, 1]>: tensor<2xi64>,
+        wavevector=dense<[0, 1]>: tensor<2xi64>>}
+        ion.pulse(%arg0: f64) %arg1 {beam=#ion.beam<
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
+        detuning=11.11,
+        polarization=dense<[0, 1]>: tensor<2xi64>,
+        wavevector=dense<[0, 1]>: tensor<2xi64>>}
+        ion.yield %arg1: !quantum.bit
+    }
+    return %2: !quantum.bit
+}
+
+func.func @example_parallel_protocol_two_qubits(%arg0: f64) -> (!quantum.bit, !quantum.bit) {
+    %0 = quantum.alloc( 1) : !quantum.reg
+    %1 = quantum.extract %0[0] : !quantum.reg -> !quantum.bit
+    %2 = quantum.extract %0[1] : !quantum.reg -> !quantum.bit
+    %3:2 = ion.parallelprotocol(%1, %2): !quantum.bit, !quantum.bit {
+        ^bb0(%arg1: !quantum.bit, %arg2: !quantum.bit):
+        ion.pulse(%arg0: f64) %arg1 {beam=#ion.beam<
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
+        detuning=11.11,
+        polarization=dense<[0, 1]>: tensor<2xi64>,
+        wavevector=dense<[0, 1]>: tensor<2xi64>>}
+        ion.pulse(%arg0: f64) %arg2 {beam=#ion.beam<
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
+        detuning=11.11,
+        polarization=dense<[0, 1]>: tensor<2xi64>,
+        wavevector=dense<[0, 1]>: tensor<2xi64>>}
+        ion.yield %arg1, %arg2: !quantum.bit, !quantum.bit
+    }
+    return %3#0, %3#1: !quantum.bit, !quantum.bit
+}
+
+
+func.func @example_ion() -> !ion.ion {
+    %0 = ion.ion {
```

Comment:
I'm lacking some context so feel free to resolve this, but why do we need a value semantic for an ion? A assume they represent a qubit? So the ultimate goal is to lower to an IR with pulses acting on ions? 

I guess the question is, does the OQD backend API expect pulse on qubits, or pulse on ions? If the latter, do we know how to map our abstract qubit types (representing a textbook logical qubit) to their physical ions?